src/anemoi/training/commands/profiler.py

```diff
@@ -0,0 +1,55 @@
+# (C) Copyright 2024 Anemoi contributors.
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+
+
+from __future__ import annotations
+
+import logging
+import os
+import sys
+
+from anemoi.training.commands.train import Train
+
+LOGGER = logging.getLogger(__name__)
+
+
+class Profiler(Train):
```

Comment:
I think inheriting from a common class is a good idea here. But we should avoid inheriting from concrete classes.

Also, like you already saw, you have to define the `run` in the Profiler anyway because it calls the `main()` at the end. If you don't define it, it will call the main from the train command. 

So with both these points in mind, I would suggest the following:

1. In the train command, add a `TrainBase` class that inherits from `Command`. `_merge_sysargv` can go into here.
2. `Train` and `Profiler` become subclasses of `TrainBase`
3. Move lines 28-37 of profiler.py into a new function in `TrainBase`. You can call it something like `prepare_sysargv` 
4. The `run` in `Profiler` and `Train` then become:
```python
def run(self, ...):
    self.prepare_sysargv()
    LOGGER.info(...)
    main()
```

There is still a little duplication because you have to define `run` in both subclasses, but it is minimal.

Reply:
Thanks for the feedback Gert! as discussed offline this is now implemented. I also reviewed the naming of the command. Right now, the command is 'anemoi-training profiler'. While we could 'rename' it to 'anemoi-training profile' to make it more consistent with 'anemoi-training train'. This is not possible due to conflicts with out pre-commit hooks `"src/anemoi/training/commands/profile.py:1:1: A005 Module `profile` is shadowing a Python builtin module"`  (no file in the codebase can then be called profile.py).
I would then propose to leave it as 'profiler' unless someone else knows a better way to overcome this Ruff